### DISCUSSION TITLE
Future maintenance of p2p protocols

### DISCUSSION THREAD

**Discussion by fallen-icarus** [fallen-icarus/cardano-swaps#8] [General]
Currently, I am the sole maintainer of the p2p protocols - all code is located in my repos and are under my licenses. While others can always fork the code and release their own versions, the beacon tokens necessitate that users agree to use specific versions of each protocol. This consensus must be updatable and the code in use must be easy to verify. 

Questions:
1) Is it okay for me to effectively be the BDFL of these protocols or should there be another method?
2) Where should the source code be located? Is my repo good enough?
3) How should the licenses be managed?

**Comment by fallen-icarus** [comment]
@rphair I thought it would be better to move the discussion out of the redundant executions PR. I would like to respond to your [comment](https://github.com/cardano-foundation/CIPs/pull/418#issuecomment-1683441424).

>> Will the official location for the code be my repo, the CIPs repo, or an entirely new repo like a "Cardano Protocols" repo?

> So far we've determined that CIP related code (practical & reference implementations; examples are OK) should not be in the CIPs repo for a few reasons highlighted in other PRs:
>
> - code changes (even for instructive examples) require PRs that have to funnel through editors who might not be the best to approve or improve them.
>- it's easier to subscribe, star, file issues, etc. for a project in its own repository that blur this development with the CIP repo divided among a lot of different documentation efforts.
>- the CIP process is focused on documentation rather than development, in addition to other subjective reasons.

I agree with the above points about using the CIP repo. However, I also think my personal repo is not the right choice. I lean towards a dedicated repo owned by some organization. Considering most open source code has some foundation backing it (eg, linux foundation, rust foundation, etc), I lean towards the official repo being located under the cardano foundation. I would be more than happy to be a maintainer of the protocols no matter where they are located and, ideally, it should not be just me.

>> Should an entity like the cardano foundation hold the license to them?

> I can think of no practical goal that could be achieved any more easily by making that assignment. Anything the community does to promote Cardano P2P markets can rely entirely upon open source licenses published by developers themselves... and anything the CF does to promote these would not depend upon its holding license over either the code or its documentation in any CIPs.

My concern about license management is legal, not promotional. I feel like I am taking on a lot of legal risk by having the official code under my name. The cardano foundation is much better positioned to handle this risk than an individual such as myself.

**Reply by rphair** [reply]
Yes, I see what you mean now.  I apologise for not thinking in terms of your own risk; especially given the illegal and unjustifiable detainments and confiscations over the Tornado protocol.

We should seriously consider _all_ the points you have made and I believe this is a good candidate to bring up at our next CIP meeting, which is scheduled for the 29th (4PM UTC).  You would be warmly welcome at that meeting (agenda: https://hackmd.io/@cip-editors/72)

CIP Editors from both the CF (@KtorZ) and IOG (@Ryun1) will be there... not representing these organisations in such capacity, but able to provide some insight... and I will introduce the dialogue so far & we can discuss how to more officially request more formal support in these areas.

**Reply by rphair** [reply]
p.s. OK, now you can see the last comment I made on this CIP meeting agenda; we can agree among editors how to best document the CIP-related factors so the more general issues are not buried in the specific Plutus difficulties of the draft CPS-0004.  The first question of course would be whether & how the maintenance & legal matters might be discussed outside the CIP Editing scope.
